### DISCUSSION TITLE
How does reporting and timezones interact

### DISCUSSION THREAD

**Discussion by seanvl** [kimai/kimai#2199] [General]
I really like Kimai but I am having some trouble understanding the interactions with timezones. In kimai2_timesheet table the start/end times are set to times for PST and the timezone is set to America/Los_Angeles.

When looking at data in the user interface from the Administration/Timesheets  link in the sidebar, the data seems to be displayed in UTC. Is it possible to view this data in a particular timezone? For me that would be America/Los_Angeles? The issue is that reports for specific date ranges are shifted by 8 hours which does not line up with customer data that is in PST. It seems that the fix would be to set the reports to report for a timezone and if the data is set to that same timezone no shift would happen. I tried looking at the documentation, sorry if I missed where it details how to remedy this.

**Comment by kevinpapst** [comment]
Reports show, other than the timesheet view, the raw data - so they are displayed in their original timezone.

There is no need for a fix, converting this data to your own timezone would be a different kind of report.

**Reply by seanvl** [reply]
Thanks for the fast reply! Perhaps I am not understanding things correctly. One of the screen captures is the kimai2_timesheet table. Is this the data that you say should be reported in the timesheet view as stored? The second screen capture is for what I see on screen. Am I comparing the wrong things?
![Kimia2_timesheet screenshot](https://user-images.githubusercontent.com/4692897/102372189-ce4eed00-3f73-11eb-9e39-4e2559e193e9.png)
![capture timesheets for same user](https://user-images.githubusercontent.com/4692897/102372210-d444ce00-3f73-11eb-8405-e85823159fe7.png)

**Reply by kevinpapst** [reply]
Recorded times in the database are stored in UTC with the original timezone.
Timesheet view always show the times in your own timezone.
Reports shows the times as recorded.

**Reply by seanvl** [reply]
Thanks for the update. Are invoices counted as reports?
